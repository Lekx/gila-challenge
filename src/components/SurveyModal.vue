<script setup lang="ts">
import { ref, watch } from 'vue';
import InputText from '@/components/InputText.vue';
import InputNumber from '@/components/InputNumber.vue';
// Props defined for component:
// showSurveyModal - to display the modal or not
const props = defineProps({
    showSurveyModal: {
        type: Boolean,
        required: true,
        default: false,
    },
});

// var that holds the modal class to show or hide the modal
const modalStyleState = ref('hide');

// Watcher to listen for parent button change on showing or not the modal
watch(
    () => props.showSurveyModal,
    (newValue) => {
        modalStyleState.value = newValue ? 'show' : 'hide';
    }
);

// Arrow function to change modalStyleState value to hide in order to hide the modal
const hideSurveyModal = () => {
    modalStyleState.value = 'hide';
};

const handleInputChange = (value: string) => {
    console.log(value);
}
</script>
<template>
    <div :class="['overlay', modalStyleState]">
        <div class="modal modal-md" id="survey-modal">
            <div class="ly-row ly-spacebetween-center mb-1">
                <h1 class="h3 m-0">Supermarket Survey!</h1>
                <a href="" class="btn btn-noborder btn-icon">
                    <i class="lpi-close"></i><span>Close</span>
                </a>
            </div>
            <div class="form">
                <!-- form placeholder -->
                <div class="ly-row">
                    <InputNumber class="w-50" labelText="Select a Fruit" :max="9" type="integer" @input-change="handleInputChange" />
                    <InputText class="w-50" labelText="Select a Food item" />
                </div>
                <div class="ly-row">
                    <InputText class="w-50" labelText="Favorite Food" helperText="E.g. Seafood Pizza."
                        defaultValue="mms wey" :max="80" />
                    <InputText class="w-50" labelText="Food Budget" helperText="Solo numbers" :max="5" />
                </div>
            </div>
            <div class="modal-footer text-right">
                <a href="" class="btn hide-btn" @click="hideSurveyModal()">Cancel</a>
                <a href="" class="btn btn-primary">Submit</a>
            </div>
        </div>
    </div>
</template>
<style lang="scss" scoped>
// classes to fix the library functionality for toggling the modal
.show {
    display: flex;

    .modal {
        display: block;
    }
}

.hide {
    display: none;
}
</style>



