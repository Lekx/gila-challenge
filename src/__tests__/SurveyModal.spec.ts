import { mount } from '@vue/test-utils';
import { describe, it, expect } from 'vitest';
import SurveyModal from '../components/SurveyModal.vue';

describe('HomeView', () => {
    it('should render the correct HTML', () => {
        const wrapper = mount(SurveyModal);

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('should have a title of "Supermarket Survey!"', () => {
        const wrapper = mount(SurveyModal);

        expect(wrapper.find('h1').text()).toEqual('Supermarket Survey!');
    });

    it('should have a button with the text "Cancel"', () => {
        const wrapper = mount(SurveyModal);

        expect(wrapper.find('a.btn.hide-btn').text()).toEqual('Cancel');
    });

    it('should have a button with the text "Submit"', () => {
        const wrapper = mount(SurveyModal);

        expect(wrapper.find('a.btn.btn-primary').text()).toEqual('Submit');
    });
});
