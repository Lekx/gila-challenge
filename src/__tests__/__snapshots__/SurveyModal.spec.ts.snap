// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomeView > should render the correct HTML 1`] = `
"<div data-v-dccb68fe=\\"\\" class=\\"overlay hide\\">
  <div data-v-dccb68fe=\\"\\" class=\\"modal modal-md\\" id=\\"survey-modal\\">
    <div data-v-dccb68fe=\\"\\" class=\\"ly-row ly-spacebetween-center mb-1\\">
      <h1 data-v-dccb68fe=\\"\\" class=\\"h3 m-0\\">Supermarket Survey!</h1><a data-v-dccb68fe=\\"\\" href=\\"\\" class=\\"btn btn-noborder btn-icon\\"><i data-v-dccb68fe=\\"\\" class=\\"lpi-close\\"></i><span data-v-dccb68fe=\\"\\">Close</span></a>
    </div>
    <div data-v-dccb68fe=\\"\\" class=\\"form\\">
      <!-- form placeholder -->
      <div data-v-dccb68fe=\\"\\" class=\\"ly-row\\">
        <div data-v-f6c8fbf5=\\"\\" data-v-dccb68fe=\\"\\" class=\\"w-50\\"><label data-v-f6c8fbf5=\\"\\" class=\\"input-text d-block py-1 lead text-gray-dark\\">Select a Fruit</label>
          <div data-v-f6c8fbf5=\\"\\" class=\\"input-field-wrapper integer\\">
            <!--v-if--><input data-v-f6c8fbf5=\\"\\" class=\\"input-field py-1\\" type=\\"text\\" maxlength=\\"9\\">
          </div>
        </div>
        <div data-v-2f6f31ac=\\"\\" data-v-dccb68fe=\\"\\" class=\\"component-wrapper w-50\\"><label data-v-2f6f31ac=\\"\\" class=\\"input-text d-block py-1 lead text-gray-dark\\">Select a Food item</label>
          <div data-v-2f6f31ac=\\"\\" class=\\"input-field-wrapper\\"><i data-v-2f6f31ac=\\"\\" class=\\"lpi-caret-down\\"></i><input data-v-2f6f31ac=\\"\\" class=\\"input-field py-1\\" type=\\"text\\" disabled=\\"\\" placeholder=\\"Select one\\"></div>
          <!--v-if-->
        </div>
      </div>
      <div data-v-dccb68fe=\\"\\" class=\\"ly-row\\">
        <div data-v-dccb68fe=\\"\\" class=\\"w-50\\"><label class=\\"input-text d-block py-1 lead text-gray-dark\\">Favorite Food</label><input class=\\"input-field py-1\\" type=\\"text\\" maxlength=\\"80\\">
          <div class=\\"input-helpers ly-row\\"><span class=\\"helper-text py-1 w-80 text-dark-muted\\">E.g. Seafood Pizza.</span><span class=\\"helper-length py-1 w-20 text-right\\">7/80</span></div>
        </div>
        <div data-v-dccb68fe=\\"\\" class=\\"w-50\\"><label class=\\"input-text d-block py-1 lead text-gray-dark\\">Food Budget</label><input class=\\"input-field py-1\\" type=\\"text\\" maxlength=\\"5\\">
          <div class=\\"input-helpers ly-row\\"><span class=\\"helper-text py-1 w-80 text-dark-muted\\">Solo numbers</span><span class=\\"helper-length py-1 w-20 text-right\\">0/5</span></div>
        </div>
      </div>
    </div>
    <div data-v-dccb68fe=\\"\\" class=\\"modal-footer text-right\\"><a data-v-dccb68fe=\\"\\" href=\\"\\" class=\\"btn hide-btn\\">Cancel</a><a data-v-dccb68fe=\\"\\" href=\\"\\" class=\\"btn btn-primary\\">Submit</a></div>
  </div>
</div>"
`;
